lib/costellation.dart

```diff
@@ -0,0 +1,642 @@
+    Pointer<G_Item> item = bindings.file_into_item(_pointer);
+
+    if (item == nullptr) {
+      throw Exception("File not found");
+    }
+
+    bindings.item_set_description(
+        item, descritpion.toNativeUtf8().cast<Char>());
+
+    _pointer = bindings.item_into_file(item).data;
+
+    return File(_pointer);
+  }
+
+  File rename(String name) {
+    Pointer<G_Item> item = bindings.file_into_item(_pointer);
+
+    if (item == nullptr) {
+      throw Exception("File not found");
+    }
+
+    bindings.item_rename(item, name.toNativeUtf8().cast<Char>());
+
+    _pointer = bindings.item_into_file(item).data;
+
+    return File(_pointer);
+  }
+
+  void drop() {
+    bindings.file_free(_pointer);
+  }
+}
+
+class Directory {
+  late Pointer<G_Directory> _pointer;
+  late String id;
+  late String name;
+  late String description;
+  late DateTime creation;
+  late DateTime modified;
+
+  Directory(this._pointer) {
+    Pointer<Char> pId = bindings.directory_id(_pointer);
+    id = pId.cast<Utf8>().toDartString();
+    Pointer<Char> pName = bindings.directory_name(_pointer);
+    name = pName.cast<Utf8>().toDartString();
+    Pointer<Char> pDescription = bindings.directory_description(_pointer);
+    description = pDescription.cast<Utf8>().toDartString();
+    int pCreation = bindings.directory_creation(_pointer);
+    creation = DateTime.fromMillisecondsSinceEpoch(pCreation);
+    int pModified = bindings.directory_modified(_pointer);
+    modified = DateTime.fromMillisecondsSinceEpoch(pModified);
+
+    calloc.free(pId);
+    calloc.free(pName);
+    calloc.free(pDescription);
+  }
+
+  void addDirectory(Directory directory) {
+    G_FFIResult_Null result =
+        bindings.directory_add_directory(_pointer, directory.pointer());
+
+    if (result.error != nullptr) {
+      throw WarpException(result.error);
+    }
+    bindings.free(result.data);
+  }
+
+  bool hasItem(Directory directory) {
+    Pointer<Char> item = directory.name.toNativeUtf8().cast<Char>();
+
+    bool has_item = bindings.directory_has_item(_pointer, item);
+
+    if (has_item == 0) {
+      return false;
+    }
+    calloc.free(item);
+    return true;
+  }
+
+  void addFile(File file) {
+    G_FFIResult_Null result =
+        bindings.directory_add_file(_pointer, file._pointer);
+
+    if (result.error != nullptr) {
+      throw WarpException(result.error);
+    }
+    bindings.free(result.data);
+  }
+
+  void addItem(Item item) {
+    G_FFIResult_Null result =
+        bindings.directory_add_item(_pointer, item.pointer());
+
+    if (result.error != nullptr) {
+      throw WarpException(result.error);
+    }
+    bindings.free(result.data);
+  }
+
+  Item directoryToItem() {
+    Pointer<G_Item> result = bindings.directory_into_item(_pointer);
+
+    if (result == nullptr) {
+      throw Exception("File not found");
+    }
+    Item item = Item(result);
+    bindings.item_free(result);
+    return item;
+  }
+
+  Directory setDescription(String descritpion) {
```

Comment:
Ditto but with returning `Directory`

Reply:
Now I simply replace the pointer already stored and added a function to get the most update file pointer